plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/ProjectRepository.java

```diff
@@ -103,7 +96,32 @@ private static Projects getProjectsApi(Credential credential) {
   }
 
   private static GcpProject convertToGcpProject(Project project) {
+    Preconditions.checkNotNull(project);
     return new GcpProject(project.getName(), project.getProjectId());
   }
 
+  /**
+   * @return true if the credential has access to the GCP project identified by {@code projectId}
+   *     and the project has an App Engine application
+   * @throws ProjectRepositoryException if an error other than HTTP 404 happens while retrieving the
+   *     App Engine application
+   */
+  public boolean hasAppEngineApplication(Credential credential, String projectId)
+      throws ProjectRepositoryException {
+    Preconditions.checkNotNull(credential);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+
+    try {
+      apiFactory.getAppsApi(credential).get(projectId).execute();
```

Comment:
Can we also check the return value, just in case they change what they throw as an exception in the future?